test/metabase/query_processor/streaming/xlsx_test.clj

```diff
@@ -748,3 +753,27 @@
                           {}
                           [[1]
                            [2]]))))))
+
+(deftest pivot-table-resource-usage-test
+  (testing "pivot table initialization should complete in reasonable time and memory"
+    ;; We test XLSX export of an empty table (0 rows) with pivoting enabled. This should test the initialization of
+    ;; pivot machinery that used to allocate and retain a lot of memory (and hence was slow on smaller heaps).
+    (let [do-export #(xlsx-export [{:display_name "A"}
+                                   {:display_name "B"}
+                                   {:display_name "C"}
+                                   {:display_name "D"}
+                                   {:display_name "pivot-grouping"}
+                                   {:display_name "E"}
+                                   {:display_name "F"}]
+                                  {}
+                                  []
+                                  :pivot-export-options {:pivot-rows [0 1], :pivot-cols [0 1], :pivot-measures [5 4]})
+          ;; Run once before measuring to warm-up and thus reduce flakiness.
+          _ (do-export)
+          start (System/nanoTime)
+          start-bytes (get-allocated-bytes)]
+      (time (do-export))
+      ;; Should always be faster than 1 second. If this test ever flakes, consider tuning or disabling.
+      (is (< (- (System/nanoTime) start) (* 1 1000 1000 1000)))
+      ;; Should always allocate less than 100Mb.
+      (is (< (- (get-allocated-bytes) start-bytes) (* 100 1024 1024))))))
```

Comment:
Do these assertions fail prior to your fix in this PR? (and/or is the OOM reproducible?)

As you noted in the comment, a time-based test assertion makes me nervous about flaking. If this fix results in a substantial timing difference, and it's now much less than 1 second, I guess it's fine. But if the memory allocation check reliably detects this issue, maybe that's enough?

Reply:
yea, I think let's take the timing assertion out. 
The whole purpose of the fix is related not to timing but memory use, so that assertion is the critical one.

Prior to this change, the pivot table initialization was pretty consistently over 100mb, and in the failing case reported by the linked issue, it was around 800Mb (oof!), so this is a clear improvement.

Reply:
As for poi-ooxml-full:

It's needed for the xml schema defs. If I don't include it, then modifying the pivot-table ref fails with a class not found exception. Annoying, but the quickest way to solve the issue.

I thought that poi-ooxml-full was also a superset, but when I don't also include the poi-ooxml, I get this:

```
Execution error (ClassNotFoundException) at java.net.URLClassLoader/findClass (URLClassLoader.java:445).
org.apache.poi.xssf.usermodel.XSSFWorkbook
```